pkg/catalog/routes.go

```diff
@@ -133,19 +133,21 @@ func hostnamesTostr(hostnames []string) string {
 // GetHostnamesForService returns the hostnames for a service.
 // The hostname can be the FQDN for the service, and can include ports as well.
 // Ex. bookstore.default, bookstore.default:80, bookstore.default.svc, bookstore.default.svc:80 etc.
-func (mc *MeshCatalog) GetHostnamesForService(meshService service.MeshService) (string, error) {
+func (mc *MeshCatalog) GetHostnamesForService(meshService service.MeshService) ([]string, error) {
 	log.Trace().Msgf("Finding domain for service %s", meshService)
+	var svcHostnames []string
 
 	if mc.configurator.IsPermissiveTrafficPolicyMode() {
 		hostnames, err := mc.getServiceHostnames(meshService)
 		if err != nil {
 			log.Error().Err(err).Msgf("Error getting service hostnames for MeshService %s", meshService)
-			return "", err
+			return svcHostnames, err
 		}
-		return hostnamesTostr(hostnames), nil
+		return []string{hostnamesTostr(hostnames)}, nil
```

Comment:
This seems hacky because it is going to return something like []string{"foo,bar,baz"}. Can we avoid this?